assets/src/components/selection/event-select/index.js

```diff
@@ -172,8 +182,9 @@ const whereConditions = ( { showExpired, categorySlug, month } ) => {
  * eventespresso/lists store.
  */
 export default withSelect( ( select, ownProps ) => {
-	const { limit, order, orderBy } = ownProps.attributes;
-	const where = whereConditions( ownProps.attributes );
+	const { attributes = EventSelect.defaultProps.attributes } = ownProps;
```

Comment:
is this kinda like double assigning the incoming `ownProps` variable to both `EventSelect.defaultProps.attributes` AND the `attributes` constant?

If so, then are there any actual *default* attributes saved to `EventSelect.defaultProps.attributes` that would be missed if you go on to only use the  `attributes` constant?

for example:
if `EventSelect.defaultProps.attributes` starts as `{ 'a' = 'a', 'b' = 'b', 'c' = 'c' }` 
but then the incoming `ownProps` is `{ 'a' = 'A', 'b' = 'B' }`,
then  the `attributes` constant will also be `{ 'a' = 'A', 'b' = 'B' }` and not have a value for `'c'` 

therefore should `ownProps` be merged with `EventSelect.defaultProps.attributes` before assigning to the `attributes` constant?

Reply:
Well the tricky thing here is that attributes is actually something that should be defined by the parent block using this component.  If the parent block doesn't specify attributes then they will not be persisted with the block.  So in some ways its _good_ if this errors for any properties that aren't included.  The only reason why I set the default from the default properties of the EventSelect is to make it easier to initialize the EventSelect via `<EventSelect />` in the parent block.  However even then, at a minimum there would need to be passed through the onSelect callback etc.